### PR TITLE
Pool tx replacement by nonce fix

### PR DESCRIPTION
**Pull request by tbssajal**

### CODE REVIEW

File: src/Lachain.Core/Blockchain/Error/OperatingError.cs [hunk 1]
```diff
@@ -27,5 +27,6 @@ public enum OperatingError : byte
         BlockGasOverflow,
         Underpriced,
         ValueOverflow,
+        BadTransaction,
```
Comment (sgladkov) [inline src/Lachain.Core/Blockchain/Error/OperatingError.cs:30]: I think it is better to make more narrow error type here,  smth like DuplicatedTransaction or smth like this

File: src/Lachain.Core/Blockchain/Pool/TransactionPool.cs [hunk 2]
```diff
@@ -219,15 +219,32 @@ public OperatingError Add(TransactionReceipt receipt, bool notify = true)
...
+                        Logger.LogTrace(
+                            $"Transaction {receipt.Hash.ToHex()} with nonce: {receipt.Transaction.Nonce} and gasPrice: " +
+                            $"{receipt.Transaction.GasPrice} trying to replace transaction {oldTx.Hash.ToHex()} with nonce: " +
+                            $"{oldTx.Transaction.Nonce} and gasPrice: {oldTx.Transaction.GasPrice} but cannot due to lower gasPrice");
                         return OperatingError.Underpriced;
                     }
+
+                    var identicalTx = new Transaction(receipt.Transaction);
+                    identicalTx.GasPrice = oldTx.Transaction.GasPrice;
+                    if (!identicalTx.Equals(oldTx.Transaction))
```
Comment (sgladkov) [inline src/Lachain.Core/Blockchain/Pool/TransactionPool.cs:231]: I think it is better to compare From,  To Amount and Invocation explicitly,  may be inside some separate function

File: src/Lachain.Core/Blockchain/Error/OperatingError.cs (hunk 1 above)
Comment (tbssajal) [reply src/Lachain.Core/Blockchain/Error/OperatingError.cs:30]: Ok

File: src/Lachain.Core/Blockchain/Pool/TransactionPool.cs (hunk 2 above)
Comment (sgladkov) [reply src/Lachain.Core/Blockchain/Pool/TransactionPool.cs:231]: From,  To,  Value, Invocation : 4 fields

File: src/Lachain.Core/Blockchain/Pool/TransactionPool.cs [hunk 3]
```diff
@@ -260,6 +275,12 @@ private bool IsBalanceValid(TransactionReceipt receipt)
             return balance.CompareTo(fee) >= 0;
         }
 
+        private bool IdenticalTransaction(Transaction oldTx, Transaction newTx)
+        {
+            return oldTx.From.Equals(newTx.From) && oldTx.To.Equals(newTx.To) &&
```
Comment (sgladkov) [inline src/Lachain.Core/Blockchain/Pool/TransactionPool.cs:280]: Value field is missed

**Review by sgladkov — APPROVED** [review]

**Pull request merged**